File: python-sdk/src/formanceapi/models/shared/connectorsresponse.py
import dataclasses
from ..shared import connector_enum as shared_connector_enum
from dataclasses_json import dataclass_json
from formanceapi import utils
from typing import Optional


@dataclass_json
@dataclasses.dataclass
class ConnectorsResponseData:
    enabled: Optional[bool] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('enabled') }})
    provider: Optional[shared_connector_enum.ConnectorEnum] = dataclasses.field(default=None, metadata={'dataclasses_json': { 'letter_case': utils.field_name('provider') }})
    

@dataclass_json
@dataclasses.dataclass
class ConnectorsResponse:
    data: list[ConnectorsResponseData] = dataclasses.field(metadata={'dataclasses_json': { 'letter_case': utils.field_name('data') }})
    

File: ts-sdk/src/sdk/models/shared/connectorsresponse.ts
import { SpeakeasyBase, SpeakeasyMetadata } from "../../../internal/utils";
import { ConnectorEnum } from "./connectorenum";


export class ConnectorsResponseData extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=enabled" })
  enabled?: boolean;

  @SpeakeasyMetadata({ data: "json, name=provider" })
  provider?: ConnectorEnum;
}

export class ConnectorsResponse extends SpeakeasyBase {
  @SpeakeasyMetadata({ data: "json, name=data", elemType: ConnectorsResponseData })
  data: ConnectorsResponseData[];
}